examples/workflows/experimental/new_decorators_basic_script.py

```diff
@@ -0,0 +1,27 @@
+from hera.shared import global_config
+from hera.workflows import RunnerInput, RunnerOutput, WorkflowTemplate
+
+global_config.experimental_features["script_annotations"] = True
+global_config.experimental_features["script_pydantic_io"] = True
+
+w = WorkflowTemplate(name="my-template")
+
+
+class MyInput(RunnerInput):
+    user: str
+
+
+@w.script()
+def hello_world(my_input: MyInput) -> RunnerOutput:
+    output = RunnerOutput()
+    output.result = f"Hello Hera User: {my_input.user}!"
+    return output
+
+
+# Pass script kwargs (including an alternative public template name) in the decorator
+@w.set_entrypoint()
+@w.script(name="goodbye-world", labels={"my-label": "my-value"})
```

Comment:
```suggestion
@w.script(name="goodbye-world", labels={"my-label": "my-value"}, entrypoint=True)
```

Reply:
We can either do this or the other one.

Reply:
Just wanted to comment that even though we're going with the `set_entrypoint` fix, I don't think adding this in the `script` decorator makes sense as it's a property of the workflow, not an individual template. Plus it would mess up the `kwargs` type hinting unless we added it to a mixin (except the previous point means doing that doesn't make sense).